### ISSUE TITLE
Update Weather to latest API

### ISSUE THREAD

**Issue by dingwilson** [IBM-Swift-Sunset/swift-watson-sdk#30]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_
_Originally opened as https://github.ibm.com/IBM-Swift/swift-watson-sdk/issues/30_

----

<a href="https://github.com/vherrin"><img src="https://avatars3.githubusercontent.com/u/2422521?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [vherrin](https://github.com/vherrin)**
_Monday Aug 08, 2016 at 13:30 GMT_
_Originally opened as https://github.com/IBM-Swift/swift-watson-sdk/issues/30_

----

As a consumers of the sdk, it would be nice to have access to all the new endpoints now available since there are many more added.

**Comment by dingwilson** [comment]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_

----

<a href="https://github.com/rfdickerson"><img src="https://avatars2.githubusercontent.com/u/1310293?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [rfdickerson](https://github.com/rfdickerson)**
_Saturday Aug 13, 2016 at 14:36 GMT_

----

I was wondering if we would want the Weather API to exist separately from the rest of the Watson APIs? I want to use it for a series of tutorials and demos.

**Comment by dingwilson** [comment]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_

----

<a href="https://github.com/vherrin"><img src="https://avatars3.githubusercontent.com/u/2422521?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [vherrin](https://github.com/vherrin)**
_Saturday Aug 13, 2016 at 14:47 GMT_

----

This is a great question and yes it should be pulled out at some point.  The Weather API's are now much more robust so a new SDK could be created.  Once that new SDK is complete then the older weather items in this SDK can be deprecated / removed.  Also, the older API's will no longer function after a certain period of time anyways.  Insights for Weather is no longer available.

**Comment by dingwilson** [comment]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_

----

<a href="https://github.com/kurtismgit"><img src="https://avatars0.githubusercontent.com/u/7569099?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kurtismgit](https://github.com/kurtismgit)**
_Wednesday Aug 17, 2016 at 00:45 GMT_

----

Looks like this sdk is currently using the deprecated weather API:  https://twcservice.mybluemix.net/rest-api-deprecated/

I can't seem to get that API to work with newer versions of the BlueMix weather service.  The new service name is 'Weather Company Data for IBM BlueMix'.  When attempting to use this SDK which uses that API, I get a 401 response when calling insightsForWeather.getCurrentForecast.

Are others able to successfully use this SDK with the new 'Weather Company Data for IBM BlueMix' service instances?

**Comment by dingwilson** [comment]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_

----

<a href="https://github.com/vherrin"><img src="https://avatars3.githubusercontent.com/u/2422521?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [vherrin](https://github.com/vherrin)**
_Wednesday Aug 17, 2016 at 16:15 GMT_

----

Thanks for the reply!  Yes we plan on on updating the API to reflect the updates to the service.  The deprecated version will work for 90 days from announcement with services already in place.  You cannot create a new instance.

**Comment by dingwilson** [comment]
<a href="https://github.ibm.com/Wilson-Ding1"><img src="https://avatars.github.ibm.com/u/88793?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Wilson-Ding1](https://github.ibm.com/Wilson-Ding1)**
_Thursday Sep 28, 2017 at 04:13 GMT_

----

<a href="https://github.com/kurtismgit"><img src="https://avatars0.githubusercontent.com/u/7569099?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kurtismgit](https://github.com/kurtismgit)**
_Wednesday Aug 17, 2016 at 17:33 GMT_

----

So basically that's saying this API doesn't work with the current weather service that's available in the BlueMix catalog.  Seems like something that should be called out in the readme and perhaps communicated with teams you know are using this API.

I found this out the hard way.  Our team was attempting to use the [BluePic](https://github.com/IBM-Swift/BluePic) sample app. Which depends on this SDK.  After digging through the sample code, I finally figured out it was broken because this SDK doesn't work with the new BlueMix weather service.  I opened an [issue against BluePic](https://github.com/IBM-Swift/BluePic/issues/307) project to let them know this.
